platform/platform_tests_support/scoped_file.hpp

```diff
@@ -18,7 +18,17 @@ class ScopedDir;
 class ScopedFile
 {
 public:
-  explicit ScopedFile(string const & relativePath);
+  enum Mode
+  {
+    // Create or overwrite the file and remove it at scope exit.
+    MODE_CREATE,
+
+    // Remove the file at scope exit. The caller must
+    // ensure that the file has been created by that time.
+    MODE_DO_NOT_CREATE
+  };
+
+  ScopedFile(string const & relativePath, Mode mode);
 
   ScopedFile(string const & relativePath, string const & contents);
```

Comment:
Please, describe semantic of the method here.

Reply:
done